lib/utils/azure_queues.ts

```diff
@@ -127,3 +135,68 @@ export function updateMessageVisibilityTimeout<T extends IQueueMessage>(
       });
   });
 }
+
+/**
+ * Returns false when we need to trigger a retry.
+ */
+export async function handleQueueProcessingFailure(
+  queueService: QueueService,
+  queueMessage: IQueueMessage,
+  queueName: string,
+  onTransientError: () => Promise<Either<RuntimeError, {}>>,
+  onPermanentError: () => Promise<Either<RuntimeError, {}>>,
+  error: Error | RuntimeError
+): Promise<void> {
+  const runtimeError = RuntimeErrorOf(error);
```

Comment:
by looking at this code it's really hard to guess what `RuntimeErrorOf` is supposed to do

Reply:
It translates an upcoming Error (or anything that may be throwed inside the code) into a RuntimeError